AirCasting/MicrophoneSession/MicrophoneManager.swift

```diff
@@ -21,86 +18,110 @@ class MicrophoneManager: ObservableObject {
     private(set) var isRecording = false
     
     private var recorder: AVAudioRecorder!
+    private(set) var audioSession = AVAudioSession.sharedInstance()
     private var levelTimer = Timer()
     
     private var locationProvider = LocationProvider()
     
     func startRecording(session: Session) {
         if isRecording { return }
         
-        let audioSession = AVAudioSession.sharedInstance()
-        
-        //Check microphone permisson
-        if audioSession.recordPermission != .granted {
-            audioSession.requestRecordPermission { (isGranted) in
-                if !isGranted {
-                    //TODO: pop-up that informs user that we need access to mic with ability do go back to homescreen
-                    fatalError("You must allow audio recording for this demo to work")
-                }
+        if audioSession.recordPermission == .granted {
+            addInterruptionsObserver()
+            
+            self.session = session
```

Comment:
I know that at this point it might be not easy to do but this needs refactoring. Session is a NSManagedObject. Keeping references to this objects are a bad idea. They are strictly related to the NSManageObjectContext. If ever this session would be removed or edited in any other way in the background the reference may fail/ crash. Also may not be a good idea to base recording logic on a database entity. A session could be created at the begging of a recording if really needed but until recording is finished or has at least reasonable amount of data to store this should be kept in some kind of in memory only object. For future reference lets call this kind of objects PODO's (Plane old data object).

Reply:
done